packages/backend-function/src/factory.test.ts

```diff
@@ -454,6 +458,34 @@ void describe('AmplifyFunctionFactory', () => {
     });
   });
 
+  void describe('architecture property', () => {
+    void it('sets valid architecture', () => {
+      const lambda = defineFunction({
+        entry: './test-assets/default-lambda/handler.ts',
+        architecture: 'arm64',
+      }).getInstance(getInstanceProps);
+      const template = Template.fromStack(lambda.stack);
+
+      template.hasResourceProperties('AWS::Lambda::Function', {
+        Architecture: Architecture.ARM_64.name,
+      });
+    });
+  });
+
+  void it('throws on invalid architecture', () => {
+    assert.throws(
+      () =>
+        defineFunction({
+          entry: './test-assets/default-lambda/handler.ts',
+          architecture: 'invalid' as FunctionArchitecture,
+        }).getInstance(getInstanceProps),
+      new AmplifyUserError('InvalidArchitectureError', {
+        message: `Invalid function architecture of invalid`,
+        resolution: 'runtime must be one of the following: arm64, x86_64',
```

Comment:
The tests are failing, can you please check?
```suggestion
        resolution: 'architecture must be one of the following: arm64, x86_64',
```

Reply:
Thank you for spotting the typo!